packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -65,22 +65,29 @@ export default function MockSlice(
   sliceDiff?: SliceDiff | undefined
 ): ComponentMocks {
   const sliceMockConfig = buildSliceMockConfig(sliceModel, legacyMockConfig);
+
   return sliceMockConfig.map((sc) => {
     if (!sliceDiff) return SharedSliceMock.generate(sliceModel, sc);
 
     const variationMock = previousMocks?.find(
       (m) => m.variation === sc.variation
     );
-    if (!variationMock) return SharedSliceMock.generate(sliceModel, sc);
+    if (!variationMock) {
+      return SharedSliceMock.generate(sliceModel, sc);
+    }
 
     const patched = SharedSliceMock.patch(
       sliceDiff,
       sliceModel,
       variationMock,
       sc
     );
-    if (!patched.ok || !patched.result)
-      return SharedSliceMock.generate(sliceModel, sc);
+
+    if (!patched.ok || !patched.result) {
+      // odd case here where patched is false with the error "Error: The model of the content with variation default has not changed.", but here it would be regenerated
+      // return SharedSliceMock.generate(sliceModel, sc);
+      return variationMock;
```

Comment:
Are we mixing 2 cases here?
- something actually bad happened, and we should probably regenerate from scratch
- there is no diff and we don't want to change anything (and could we detect this just after line 70 and return early?)

Reply:
I could split them up :)